### DISCUSSION TITLE
Using more than 16 input channels

### DISCUSSION THREAD

**Discussion by sykhro** [FluidSynth/fluidsynth#1039] [Q&A] [answered]
For my use case, I have MIDIs with more than 16 channels.  I've tried setting "synth.midi-channels" to 256 but nothing changes. Does the midi need to use meta events like port select (`FF 21 01 xx`)?

**Comment by derselbst** [comment, marked as answer]
The MIDI protocol is inherently limited to 16 channels. `synth.midi-channels` will increase the number of channels that the synth can handle. But it cannot fix the limitation of the protocol. You either need to
* use the sequencer to dispatch events with custom channel mappings to the synth, or
* use a MIDI driver that creates multiple ports (like alsa_seq), or
* use those SysEx events as you suggested (not sure, if fluidsynth even supports them at the moment).

**Reply by sykhro** [reply]
Disregard what I said, it has just occurred to me that the channel number is limited to 4 bits. I'll have to check the files manually to see what's happening, and more importantly, figure out how this was even working back when I used DirectMusic
